pilot/pkg/bootstrap/certcontroller.go

```diff
@@ -205,6 +222,28 @@ func (s *Server) watchRootCertAndGenKeyCert(stop <-chan struct{}) {
 	}
 }
 
+func (s *Server) RotateDNSCertForK8sCA(stop <-chan struct{},
+	defaultCACertPath string,
+	signerName string,
+	approveCsr bool,
+	requestedLifetime time.Duration,
+) {
+	certUtil := certutil.NewCertUtil(int(defaultCertGracePeriodRatio * 100))
+	for {
+		waitTime, _ := certUtil.GetWaitTime(s.istiodCertBundleWatcher.GetKeyCertBundle().CertPem, time.Now(), time.Duration(0))
```

Comment:
is there any risk that GenKeyCertK8sCA somehow returns a cert that makes GetWaitTime return 0s and we spin in an infinite loop?

Reply:
I can't think of that case unless k8s CA does not behave as expected and returns expired certs.